### PR TITLE
Upload组件:  修复watch 中 直接修改props  fileList

### PR DESCRIPTION
**Pull request by yyn835314557**
修复 upload组件 直接修改props  fileList

Please make sure these boxes are checked before submitting your PR, thank you!

* [ ] Make sure you follow Element's contributing guide ([中文](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.zh-CN.md) | [English](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.en-US.md) | [Español](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.es.md)).
* [ ] Make sure you are merging your commits to `dev` branch.
* [ ] Add some descriptions and refer relative issues for you PR.

### CODE REVIEW

**Comment by iamkun** [comment]
Please fix lint error first.
`133:12  error  Missing semicolon  semi`

File: packages/upload/src/index.vue [hunk 1]
```diff
@@ -126,9 +126,11 @@ export default {
...
       handler(fileList) {
         this.uploadFiles = fileList.map(item => {
-          item.uid = item.uid || (Date.now() + this.tempIndex++);
-          item.status = item.status || 'success';
-          return item;
+          return {
+            url: item.url,
+            uid: item.uid || (Date.now() + this.tempIndex++),
+            status: item.status || 'success'
+          }
```
Comment (island205) [inline packages/upload/src/index.vue]: 这里不符合我们的代码规范，漏了一个 `;`

**Comment by island205** [comment]
@yyn835314557 是否可以提供一个复现问题的例子，或者添加一个 测试用力

**Comment by island205** [comment]
PR without descriptions and refer relative issues  will be closed directly.

**Pull request closed without merging**